src/System Application/App/AI/src/Azure OpenAI/AOAIDeploymentsImpl.Codeunit.al

```diff
@@ -103,12 +109,44 @@ codeunit 7769 "AOAI Deployments Impl"
         exit(GetDeploymentName(GPT4oMiniLatestLbl, CallerModuleInfo));
     end;
 
+    local procedure InitializeDeploymentDeprecationDates()
+    begin
+        if DeprecationDatesInitialized then
+            exit;
+
+        // Add deprecated deployments with their deprecation dates here
+        DeploymentDeprecationDates.Add(Turbo0301SaasLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(GPT40613SaasLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(Turbo0613SaasLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(GPT35TurboLatestLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(GPT35TurboPreviewLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(GPT4PreviewLbl, DMY2Date(1, 11, 2024));
+        DeploymentDeprecationDates.Add(GPT4LatestLbl, DMY2Date(1, 11, 2024));
```

Comment:
What about using a date variable instead?
If you do not intend to update the year later, it could also be set using the 20241101D constant.

Reply:
The models are supposed to be deprecated individually with potentially different dates. An initial cutoff date in the past has been chosen for models already deprecated.

Reply:
Use of constants would work, readability was prioritized